apollo-federation/src/sources/connect/models/validation.rs

```diff
@@ -514,6 +569,169 @@ fn parse_url(value: &Node<Value>, coordinate: &str, sources: &SourceMap) -> Resu
     Ok(url)
 }
 
+fn validate_header_arg<'a>(
+    directive_name: &Name,
+    argument_name: &str,
+    headers: &'a Node<Value>,
+    source_map: &SourceMap,
+    object: Option<&Name>,
+    field: Option<&Name>,
+) -> Result<&'a Node<Value>, Vec<Message>> {
+    let mut errors = Vec::new();
+
+    headers
+        .as_list()
+        .map(|l| l.iter().filter_map(|o| o.as_object()).collect_vec())
+        .unwrap_or_else(|| headers.as_object().map(|o| vec![o]).unwrap_or_default())
```

Comment:
Can `headers` be an object? I thought it had to be a list of objects

Reply:
i just tested this — we're working with the AST which doesn't have semantic understanding of definitions. it's valid to provide a single value instead of a list for a list argument, so we have to handle both cases. Here's an example AST 
```
[apollo-federation/tests/main.rs:22:5] &field.directives = [
    89..106 @1 Directive {
        name: "myDirective",
        arguments: [
            92..105 @1 Argument {
                name: "listOfStrings",
                value: 95..105 @1 List(
                    [
                        96..99 @1 String(
                            "a",
                        ),
                        101..104 @1 String(
                            "b",
                        ),
                    ],
                ),
            },
        ],
    },
    107..117 @1 Directive {
        name: "myDirective",
        arguments: [
            110..116 @1 Argument {
                name: "listOfStrings",
                value: 113..116 @1 String(
                    "c",
                ),
            },
        ],
    },
]
```

Reply:
I didn't know it was legal in GraphQL to provide a single value to an input that takes a list. What a wild place.